### PR TITLE
Fix transport impl

### PR DESCRIPTION
**Pull request by hihigupt**

### CODE REVIEW

File: demo/Common/mimxrt1060/transport/using_mbedtls/using_mbedtls.c [hunk 1]
```diff
@@ -801,39 +798,49 @@ int32_t TLS_FreeRTOS_send( NetworkContext_t * pNetworkContext,
...
+        {
+            /* Sent data, so update the tally and keep looping. */
+            xWritten += ( size_t ) lResult;
+        }
+        else if( ( lResult == 0 ) || ( lResult == -pdFREERTOS_ERRNO_ENOSPC ) )
+        {
+            /* No data sent. The secure sockets
+            * API supports non-blocking send, so stop the loop but don't
+            * flag an error. */
+        	lResult = 0;
```
Comment (jspaith) [inline demo/Common/mimxrt1060/transport/using_mbedtls/using_mbedtls.c:824]: 1 - I'm concerned on this - we break out of loop but we return to caller that we've written the `xWritten` which may not be accurate.  Is the socket guaranteed to write *all* the data or can it chose to just write some?

2 - Another question is on buffer lifetime management for caller?  After we return from `TLS_FreeRTOS_send` will the caller assume it doesn't need to keep buffer around anymore?  Will a non-blocking I/O make a copy of that buffer so this is OK, or is there risk of freeing?

3 - Also the `-pdFREERTOS_ERRNO_ENOSPC` - putting a negative in front of an error code - isn't a convention I've ever seen.  Please add a comment about what's up with this.

4 - My ;tl-dr here maybe should be "Can we (or should we) flip on blocking I/O and what happens if we do other than potentially some perf hit?"

File: demo/Common/mimxrt1060/transport/using_mbedtls/using_mbedtls.c [hunk 2]
```diff
@@ -759,40 +759,37 @@ int32_t TLS_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
...
-        /* Mark these set of errors as a timeout. The libraries may retry read
-         * on these errors. */
-        tlsStatus = 0;
-    }
-    else if( tlsStatus < 0 )
-    {
-        LogError( ( "Failed to read data: mbedTLSError= %s : %s.",
-                    mbedtlsHighLevelCodeOrDefault( tlsStatus ),
-                    mbedtlsLowLevelCodeOrDefault( tlsStatus ) ) );
+        lResult = ( int ) xRead;
```
Comment (danewalton) [inline demo/Common/mimxrt1060/transport/using_mbedtls/using_mbedtls.c:783]: ?
```suggestion
        lResult = ( int32_t ) xRead;
```

File: demo/Common/stm32h745/transport/using_mbedtls/using_mbedtls.c [hunk 3]
```diff
@@ -759,40 +759,37 @@ int32_t TLS_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
...
-        /* Mark these set of errors as a timeout. The libraries may retry read
-         * on these errors. */
-        tlsStatus = 0;
-    }
-    else if( tlsStatus < 0 )
-    {
-        LogError( ( "Failed to read data: mbedTLSError= %s : %s.",
-                    mbedtlsHighLevelCodeOrDefault( tlsStatus ),
-                    mbedtlsLowLevelCodeOrDefault( tlsStatus ) ) );
+        lResult = ( int ) xRead;
```
Comment (danewalton) [inline demo/Common/stm32h745/transport/using_mbedtls/using_mbedtls.c:783]:
```suggestion
        lResult = ( int32_t ) xRead;
```

File: demo/Common/stm32l4_l4+/transport/using_mbedtls/using_mbedtls.c [hunk 4]
```diff
@@ -759,40 +759,37 @@ int32_t TLS_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
...
-        /* Mark these set of errors as a timeout. The libraries may retry read
-         * on these errors. */
-        tlsStatus = 0;
-    }
-    else if( tlsStatus < 0 )
-    {
-        LogError( ( "Failed to read data: mbedTLSError= %s : %s.",
-                    mbedtlsHighLevelCodeOrDefault( tlsStatus ),
-                    mbedtlsLowLevelCodeOrDefault( tlsStatus ) ) );
+        lResult = ( int ) xRead;
```
Comment (danewalton) [inline demo/Common/stm32l4_l4+/transport/using_mbedtls/using_mbedtls.c:783]:
```suggestion
        lResult = ( int32_t ) xRead;
```

**Review by danewalton — APPROVED** [review]

File: demo/Common/stm32h745/transport/using_mbedtls/using_mbedtls.c [hunk 5]
```diff
@@ -801,39 +798,49 @@ int32_t TLS_FreeRTOS_send( NetworkContext_t * pNetworkContext,
...
+        lResult = mbedtls_ssl_write( &( pTlsTransportParams->sslContext.context ),
+                                     pBuffer + xWritten,
+                                     bytesToSend - xWritten );
+
+        if( lResult > 0 )
+        {
+            /* Sent data, so update the tally and keep looping. */
+            xWritten += ( size_t ) lResult;
+        }
+        else if( ( lResult == 0 ) || ( lResult == -pdFREERTOS_ERRNO_ENOSPC ) )
```
Comment (jspaith) [inline demo/Common/stm32h745/transport/using_mbedtls/using_mbedtls.c:819]: Same feedback about RE `pdFREERTOS_ERRNO_ENOSPC ` from the function above.

**Review by jspaith — CHANGES_REQUESTED** [review]
Main concerns are around the non-blocking i/o handling; it may be cool as-is but I don't fully understand from quick read of mbed tls docs around this.

**Pull request closed without merging**